content/articles/2019-05-vroom-1-0-0.Rmarkdown

```diff
@@ -0,0 +1,450 @@
+)
+```
+
+This is most useful when a column type needs additional information, such as
+for categorical data when you know all of the levels of a factor.
+
+```{r}
+data <- vroom("flights.tsv",
+  col_types = list(dest = col_factor(levels = c("EWR", "JFK", "LGA")))
+)
+```
+
+## Speed
+
+vroom is fast, but how fast?
+We benchmarked vroom using a real world dataset of taxi trip data, with
+14.7 million rows, 11 columns. It contains a mix of numeric and textual data and has a
+total file size of 1.55 GB.
+
+    #> Observations: 14,776,615
+    #> Variables: 11
+    #> $ medallion       <chr> "89D227B655E5C82AECF13C3F540D4CF4", "0BD7C8F5B...
+    #> $ hack_license    <chr> "BA96DE419E711691B9445D6A6307C170", "9FD8F69F0...
+    #> $ vendor_id       <chr> "CMT", "CMT", "CMT", "CMT", "CMT", "CMT", "CMT...
+    #> $ pickup_datetime <chr> "2013-01-01 15:11:48", "2013-01-06 00:18:35", ...
+    #> $ payment_type    <chr> "CSH", "CSH", "CSH", "CSH", "CSH", "CSH", "CSH...
+    #> $ fare_amount     <dbl> 6.5, 6.0, 5.5, 5.0, 9.5, 9.5, 6.0, 34.0, 5.5, ...
+    #> $ surcharge       <dbl> 0.0, 0.5, 1.0, 0.5, 0.5, 0.0, 0.0, 0.0, 1.0, 0...
+    #> $ mta_tax         <dbl> 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0...
+    #> $ tip_amount      <int> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0...
+    #> $ tolls_amount    <dbl> 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 4.8, 0.0, 0...
+    #> $ total_amount    <dbl> 7.0, 7.0, 7.0, 6.0, 10.5, 10.0, 6.5, 39.3, 7.0...
+
+We performed a series of simple manipulations with each approach.
+
+  - Reading the data
+  - `print()`
+  - `head()`
+  - `tail()`
+  - Sampling 100 random rows
+  - Filtering for "UNK" payment, this is 6434 rows (0.0435% of total).
+  - Aggregation of mean fare amount per payment type.
+
+<style>
+td,th {
+  padding: 0.4em;
+}
+thead {
+border-top: 1px solid #aaa;
+border-bottom: 1px solid #aaa;
+}
+table {
+margin-left: auto;
+margin-right: auto;
+border-bottom: 1px solid #aaa;
+}
+</style>
+
+```{r benchmark_plot, fig.height = 8, fig.width=10, warning = FALSE, echo = FALSE, message = FALSE}
+taxi <- read_benchmark(system.file("bench", "taxi-times.tsv", package = "vroom"), desc)
+
+# Keep only the vroom dplyr results
+taxi <- taxi %>%
+  filter(package %in% c("readr", "data.table", "vroom\ndplyr", "read.delim")) %>%
+  mutate(package = recode(package, "vroom\ndplyr" = "vroom"))
+
+plot_benchmark(taxi, "Time to analyze taxi trip data")
+
+make_table(taxi)
+```
+
+```{r cleanup, include = FALSE}
+unlink(c(files, "flights.tsv", "flights.tsv.gz"))
+```
+
+<br/>
+
+Some things to note in the results. The initial reading is much faster in vroom
+than any other method, and most of the manipulations, such as `print()`,
+`head()`, `tail()` and `sample()` are equally fast. However because the
+character data is read lazily operations such as `filter()` and `aggregrate()`
+which need character values require additional time.
+However this cost will only occur once, after the values have been read they
```

Comment:
So why are both "aggregate" and "filter" quite wide?

(You might also rename "aggregate" to "summarise" because I keep thinking you mean the base function)